### PR TITLE
InputConfig: Clean up controller management

### PR DESCRIPTION
**Pull request by lioncash**

### CODE REVIEW

File: Source/Core/InputCommon/InputConfig.cpp [hunk 1]
```diff
@@ -79,21 +84,39 @@ bool InputConfig::LoadConfig(bool isGC)
...
 }
+
+ControllerEmu* InputConfig::GetController(int index)
+{
+	return m_controllers.at(index).get();
+}
+
+void InputConfig::ClearControllers()
+{
+	for (auto& controller : m_controllers)
```
Comment (degasus) [inline Source/Core/InputCommon/InputConfig.cpp]: Is this needed at all? The destructor of the unique_ptr should also free it.

File: Source/Core/InputCommon/InputConfig.h [hunk 2]
```diff
@@ -4,29 +4,40 @@
...
 	void SaveConfig();
 
-	std::vector<ControllerEmu*>  controllers;
+	template <typename T, typename... Args>
+	void CreateController(Args&&... args)
+	{
+		m_controllers.emplace_back(std::make_unique<T>(std::forward<Args>(args)...));
+	}
+
+	ControllerEmu* GetController(int index);
```
Comment (degasus) [inline Source/Core/InputCommon/InputConfig.h]: What do you think about:
```
template <typename T> T* GetControllerAs(int index)
{
  return static_cast<T*>(GetController(index));
}
```
Maybe it's also possible to verify the type within the base class, but I have no clue about this kind of c++ magic :/

**Comment by degasus** [comment]
LGTM but the ClearControllers() method. This PR is very worth in my opinion.

File: Source/Core/InputCommon/InputConfig.cpp (hunk 1 above)
Comment (lioncash) [inline Source/Core/InputCommon/InputConfig.cpp]: Ah, true. I'll amend that tonight.

**Pull request merged**